### PR TITLE
Authentication

### LINKED ISSUE #2: Users can log in.

### PR DESCRIPTION
**Pull request by connorlay**
resolves #2 

Users can log in, view their own profile, and log out. Users who are not in the database are notified of an invalid email/password combination.

log in, profile, log out

invalid login attempt

### CODE REVIEW

File: app/views/layouts/_navbar.html.erb [hunk 1]
```diff
@@ -12,7 +12,13 @@
     </div>
     <div class="collapse navbar-collapse" id="navbar-collapse">
       <ul class="nav navbar-nav navbar-right">
-        <li><%= link_to "Sign Up", new_user_path %></li>
+        <% if user = current_user %>
```
Comment (Fosome) [inline app/views/layouts/_navbar.html.erb]: Can probably avoid passing `user` to this partial and access `current_user` directly.

**Comment by Fosome** [comment]
great work :+1:

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -14,6 +14,10 @@ def create
     end
   end
 
+  def show
+    @user = User.find params[:id]
```
Comment (mackermedia) [inline app/controllers/users_controller.rb]: Minor nitpick: inconsistent use of parens.

File: app/views/layouts/_navbar.html.erb [hunk 3]
```diff
@@ -12,7 +12,13 @@
...
+          <li><%= link_to "#{user.name}", user %></li>
```
Comment (mackermedia) [inline app/views/layouts/_navbar.html.erb]: I think you get by with just `if current_user` and `current_user.name`

File: spec/features/user_login_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.feature "UserLogin", type: :feature do
+
+  describe "A user" do
+    let(:user)         { create :user }
+    let(:invalid_user) { build :invalid_user }
```
Comment (mackermedia) [inline spec/features/user_login_spec.rb]: `:invalid_user` must be some other defined factory? Also minor nitpick: I kinda like to see the test setup in the same file so you could do something like: `let(:invalid_user) { build(:user, email: "asdf", password: "") }`. You could also move that into the context of `submitting invalid credentials`. No super strong feels so up to you!

**Comment by mackermedia** [comment]
Sooooo :hotsprings:. Nice work! :+1:

File: spec/features/user_login_spec.rb (hunk 4 above)
Comment (connorlay) [inline spec/features/user_login_spec.rb]: I defined `:invalid_user` to encapsulate invalid user parameters for consistency across my other tests. It is the same factory used in the user model and feature specs.

**Pull request merged**